src/OnchainKitProvider.tsx

```diff
@@ -48,6 +57,46 @@ export function OnchainKitProvider({
     return onchainKitConfig;
   }, [address, apiKey, chain, config, projectId, rpcUrl, schemaId]);
 
+  // Check the React context for WagmiProvider and QueryClientProvider
+  const { providedWagmiConfig, providedQueryClient } =
+    useProviderDependencies();
+
+  const defaultConfig = useMemo(() => {
+    // IMPORTANT: Don't create a new Wagmi configuration if one already exists
+    // This prevents the user-provided WagmiConfig from being overriden
+    return (
+      providedWagmiConfig ||
+      createWagmiConfig({
+        apiKey,
+        appName: value.config.appearance.name,
+        appLogoUrl: value.config.appearance.logo,
+      })
+    );
+  }, [
+    apiKey,
+    providedWagmiConfig,
+    value.config.appearance.name,
+    value.config.appearance.logo,
+  ]);
+  const defaultQueryClient = useMemo(() => {
+    // IMPORTANT: Don't create a new QueryClient if one already exists
+    // This prevents the user-provided QueryClient from being overriden
+    return providedQueryClient || new QueryClient();
+  }, [providedQueryClient]);
+
+  // If a dependency is missing, return a context with default parent providers
+  if (!providedWagmiConfig || !providedQueryClient) {
+    return (
+      <WagmiProvider config={defaultConfig}>
```

Comment:
doesn't this leave the possibility of having two wagmiProviders or queryClientProviders if one or the other wasn't set up?

Reply:
no because I think `defaultConfig` would default to the provided one for the respective dependency

Reply:
but there might be two `WagmiProviders` you're right

Reply:
unsure if there's any adverse effects there, will do a bit more testing